.github/workflows/python-release.yml

```diff
@@ -34,7 +34,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ ubuntu-22.04, windows-2022, macos-12, macos-13, macos-14 ]
+        os: [ ubuntu-22.04, windows-2022, macos-13, macos-14 ]
```

Comment:
removed `macos-12` because its deprecated and causing the `merge` step to not run

https://github.com/kevinjqliu/iceberg-python/actions/runs/12018552791

Reply:
Good catch, I noticed that `macos-15` is also available. Maybe we can add that one as well. I checked it on my local branch, and it works over there: https://github.com/Fokko/iceberg-python/actions/runs/12033072248